lua/lspconfig/util.lua

```diff
@@ -117,7 +117,14 @@ function M.search_ancestors(startpath, func)
 end
 
 local function escape_wildcards(path)
-  return path:gsub('([%[%]%?%*])', '\\%1')
+  -- Escape all potentially problematic special characters
+  -- Covers:
+  -- - Wildcards and pattern matching: [ ] ? * $ ( ) ^ % . + - ~
+  -- - Escape character: \
+  -- - Shell special characters: ` ' " ; | & < >
+  -- - Whitespace: %s (spaces, tabs, etc.)
+  -- The gsub function replaces any matches with the same character preceded by a backslash
+  return path:gsub('([%[%]%?%*%$%(%)%^%%%.%+%-%~\\`\'";|&<>%s])', '\\%1')
```

Comment:
can this be simplified using `vim.pesc()` ? 
```suggestion
  return path:gsub('([' .. vim.pesc('[]?*$()^%.+-~') .. '\\`\'";|&<>%s])', '\\%1')
```

Reply:
nice I didn't know that!